src/msw/textctrl.cpp

```diff
@@ -2862,6 +2862,21 @@ bool wxTextCtrl::MSWSetCharFormat(const wxTextAttr& style, long start, long end)
         }
     }
 
+    if( !(style.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_NONE || style.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_SOLID ) )
```

Comment:
Please rewrite using `switch` for the same reasons as above.

Reply:
OK